contracts/MemeFactory.sol

```diff
@@ -316,14 +309,11 @@ abstract contract MemeFactory {
         // Check the unleash timestamp
         require(block.timestamp >= memeSummon.summonTime + UNLEASH_DELAY, "Cannot unleash yet");
 
-        // Wrap native token to its ERC-20 version, where applicable
```

Comment:
This is a bug. If you wrap only here, there's a race condition where scheduleForAscendance happens before and its not wrapped yet ..

For symmetry, let's always KISS and just do the exact wrapping we need

Reply:
Why? You wrap the total amount here, also the one that's going to be scheduled for ascendance. This way you don't need to wrap it there.

You can't send to ascendance before at least one token is unleashed.

Reply:
Yes but once one token is unleashed the difference in amounts leads to a messy state machine. I think it's much cleaner now.